src/main/java/org/apache/sling/jcr/repoinit/impl/NodePropertiesVisitor.java

```diff
@@ -308,6 +317,17 @@ private void setNodeProperties(String nodePath, List<PropertyLine> propertyLines
         }
     }
 
+    private boolean equalValues(Value[] a1, Value[] a2) throws ValueFormatException, IllegalStateException, RepositoryException {
+        boolean result = true;
+        Iterator<Value> it1 = Arrays.stream(a1).iterator();
+        Iterator<Value> it2 = Arrays.stream(a2).iterator();
+        // each item needs to have same value in same order
+        while(it1.hasNext() && it2.hasNext())
+            result = result && it1.next().getString().equals(it2.next().getString()); 
```

Comment:
This loop never exits once `result == false` because `it1.next()` (and `it2.next()`) is never called.

Furthermore, I would compare the value's type (just in case, because the property type comparison should already have caught this.

Note: at first I thought `Arrays.equals(a1, a2)` would do the trick. But it turns out that different `Value` implementations are compared and they implement `#equals()` in a way that only values using the same implementation are equal.